app/helpers/textual_mixins/textual_refresh_status.rb

```diff
@@ -1,20 +1,32 @@
 module TextualMixins::TextualRefreshStatus
+  include ActionView::Helpers::DateHelper
   def textual_refresh_status
-    last_refresh_status = @record.last_refresh_status.titleize
-    if @record.last_refresh_date
-      last_refresh_date = time_ago_in_words(@record.last_refresh_date.in_time_zone(Time.zone)).titleize
+    refresh_status(@record)
+  end
+
+  def textual_refresh_date
+    refresh_date(@record)
+  end
+
+  def refresh_status(record)
+    last_refresh_status = record.last_refresh_status&.titleize
+    refresh_time = record.last_refresh_date
+    if refresh_time
+      last_refresh_date = refresh_time ? time_ago_in_words(refresh_time).titleize : _(Never)
       last_refresh_status << _(" - %{last_refresh_date} Ago") % {:last_refresh_date => last_refresh_date}
     end
     {
-      :label => _("Last Refresh"),
-      :value => [{:value => last_refresh_status},
-                 {:value => @record.last_refresh_error.try(:truncate, 120)}],
-      :title => @record.last_refresh_error
+      :label  => _("Last Refresh Status"),
+      :value  => [{:value => last_refresh_status},
+                  {:value => record.last_refresh_error.try(:truncate, 120)}],
+      :title  => record.last_refresh_error,
+      :status => record.last_refresh_status&.titleize,
+      :stale  => refresh_time ? refresh_time.before?(2.days.ago) : true # checking if date is within last two days range.
```

Comment:
@Fryguy removed Time.zone on date, in line no 24, i had to use ternary operator, otherwise if data is null , stale value  is null too.

Reply:
I was suggesting remove the enture `&.in_time_zone(Time.zone)`

Reply:
Ah ok, yeah, makes sense.